jdbc-yugabytedb/src/main/java/com/yugabyte/ysql/ClusterAwareLoadBalancer.java

```diff
@@ -67,8 +69,11 @@ public String getLeastLoadedServer(List<String> failedHosts) {
     // Choose a random from the minimum list
     String chosenHost = null;
     if (minConnectionsHostList.size() > 0) {
-      Random rand = new Random();
-      chosenHost = minConnectionsHostList.get(rand.nextInt(minConnectionsHostList.size()));
+      int idx = ThreadLocalRandom.current().nextInt(0, minConnectionsHostList.size());
+      chosenHost = minConnectionsHostList.get(idx);
+    }
+    if (chosenHost != null) {
```

Comment:
can `chosenHost` be null at the point of execution?

Reply:
It is a special case of very first time when the driver has not pulled any information from the server.